diff --git a/app.d.ts b/app.d.ts
index aafcce45730fe954bdfebe186b81e11073fd2883..3d76c52acbe635b5990937d57080542c2cbfa7f4 100644
--- a/app.d.ts
+++ b/app.d.ts
@@ -1,6 +1,6 @@
-import { PluginListener } from './core';
-import { Image } from './image';
-import { Theme } from './window';
+import { PluginListener } from './core.js';
+import { Image } from './image.js';
+import { Theme } from './window.js';
 /**
  * Identifier type used for data stores on macOS and iOS.
  *
diff --git a/dpi.d.ts b/dpi.d.ts
index 7e6c1dadb59ab896824195e69d3e2ac72c562f1b..fdff378f3d17e2c2924809a868a1124b008810df 100644
--- a/dpi.d.ts
+++ b/dpi.d.ts
@@ -1,4 +1,4 @@
-import { SERIALIZE_TO_IPC_FN } from './core';
+import { SERIALIZE_TO_IPC_FN } from './core.js';
 /**
  * A size represented in logical pixels.
  * Logical pixels are scaled according to the window's DPI scale.
diff --git a/image.d.ts b/image.d.ts
index 82f31c46c3e3094822aaea9ba39cea0ded1b7c03..76078e6d137655eddff6fef12717e7b34d16162e 100644
--- a/image.d.ts
+++ b/image.d.ts
@@ -1,5 +1,5 @@
-import { Resource } from './core';
-import { NativeIcon } from './menu/iconMenuItem';
+import { Resource } from './core.js';
+import { NativeIcon } from './menu/iconMenuItem.js';
 export interface ImageSize {
     width: number;
     height: number;
diff --git a/index.d.ts b/index.d.ts
index 6111f0e200e804bb08302271e9442c9b2638264d..bd916fe2c1822ca64ba46139e337918bc418ced3 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -18,16 +18,16 @@
  *
  * @module
  */
-import * as app from './app';
-import * as core from './core';
-import * as dpi from './dpi';
-import * as event from './event';
-import * as image from './image';
-import * as menu from './menu';
-import * as mocks from './mocks';
-import * as path from './path';
-import * as tray from './tray';
-import * as webview from './webview';
-import * as webviewWindow from './webviewWindow';
-import * as window from './window';
+import * as app from './app.js';
+import * as core from './core.js';
+import * as dpi from './dpi.js';
+import * as event from './event.js';
+import * as image from './image.js';
+import * as menu from './menu.js';
+import * as mocks from './mocks.js';
+import * as path from './path.js';
+import * as tray from './tray.js';
+import * as webview from './webview.js';
+import * as webviewWindow from './webviewWindow.js';
+import * as window from './window.js';
 export { app, core, dpi, event, image, menu, mocks, path, tray, webview, webviewWindow, window };
diff --git a/menu/base.d.ts b/menu/base.d.ts
index ce3a985fa4a104000468bd7cb3422d9b7192a08c..355139ca282159ad78ebf1cfe4a5dd39bacbabb7 100644
--- a/menu/base.d.ts
+++ b/menu/base.d.ts
@@ -1,10 +1,10 @@
-import { Resource } from '../core';
-import { CheckMenuItemOptions } from './checkMenuItem';
-import { IconMenuItemOptions } from './iconMenuItem';
-import { MenuOptions } from './menu';
-import { MenuItemOptions } from './menuItem';
-import { PredefinedMenuItemOptions } from './predefinedMenuItem';
-import { SubmenuOptions } from './submenu';
+import { Resource } from '../core.js';
+import { CheckMenuItemOptions } from './checkMenuItem.js';
+import { IconMenuItemOptions } from './iconMenuItem.js';
+import { MenuOptions } from './menu.js';
+import { MenuItemOptions } from './menuItem.js';
+import { PredefinedMenuItemOptions } from './predefinedMenuItem.js';
+import { SubmenuOptions } from './submenu.js';
 export type ItemKind = 'MenuItem' | 'Predefined' | 'Check' | 'Icon' | 'Submenu' | 'Menu';
 export declare function newMenu(kind: ItemKind, opts?: MenuOptions | MenuItemOptions | SubmenuOptions | PredefinedMenuItemOptions | CheckMenuItemOptions | IconMenuItemOptions): Promise<[number, string]>;
 export declare class MenuItemBase extends Resource {
diff --git a/menu/checkMenuItem.d.ts b/menu/checkMenuItem.d.ts
index ec4aa4b714b163fe656d8a5db9cc0afe7d5a1e05..28b946aa34cd64c6e10eaa869606a736c91fa868 100644
--- a/menu/checkMenuItem.d.ts
+++ b/menu/checkMenuItem.d.ts
@@ -1,5 +1,5 @@
-import { MenuItemBase } from './base';
-import { type MenuItemOptions } from '../menu';
+import { MenuItemBase } from './base.js';
+import { type MenuItemOptions } from '../menu.js';
 /** Options for creating a new check menu item. */
 export interface CheckMenuItemOptions extends MenuItemOptions {
     /** Whether the new check menu item is enabled or not. */
diff --git a/menu/iconMenuItem.d.ts b/menu/iconMenuItem.d.ts
index 10ae6b9548803fee3275032e15ce1b46cae82d9d..ede5a65c8ce0634d1d7900246623e359baf844bf 100644
--- a/menu/iconMenuItem.d.ts
+++ b/menu/iconMenuItem.d.ts
@@ -1,6 +1,6 @@
-import { MenuItemBase } from './base';
-import { type MenuItemOptions } from '../menu';
-import { MenuIcon } from '../image';
+import { MenuItemBase } from './base.js';
+import { type MenuItemOptions } from '../menu.js';
+import { MenuIcon } from '../image.js';
 /**
  * A native Icon to be used for the menu item
  *
diff --git a/menu/menu.d.ts b/menu/menu.d.ts
index 194673cdb8e786ba854111562a0735baabe7d6a6..3db4f73eee632cf818c04cce1a036a011afc6261 100644
--- a/menu/menu.d.ts
+++ b/menu/menu.d.ts
@@ -1,12 +1,12 @@
-import { MenuItemOptions, SubmenuOptions, IconMenuItemOptions, PredefinedMenuItemOptions, CheckMenuItemOptions } from '../menu';
-import { MenuItem } from './menuItem';
-import { CheckMenuItem } from './checkMenuItem';
-import { IconMenuItem } from './iconMenuItem';
-import { PredefinedMenuItem } from './predefinedMenuItem';
-import { Submenu } from './submenu';
-import { type LogicalPosition, PhysicalPosition, Position } from '../dpi';
-import { type Window } from '../window';
-import { MenuItemBase } from './base';
+import { MenuItemOptions, SubmenuOptions, IconMenuItemOptions, PredefinedMenuItemOptions, CheckMenuItemOptions } from '../menu.js';
+import { MenuItem } from './menuItem.js';
+import { CheckMenuItem } from './checkMenuItem.js';
+import { IconMenuItem } from './iconMenuItem.js';
+import { PredefinedMenuItem } from './predefinedMenuItem.js';
+import { Submenu } from './submenu.js';
+import { type LogicalPosition, PhysicalPosition, Position } from '../dpi.js';
+import { type Window } from '../window.js';
+import { MenuItemBase } from './base.js';
 /** Options for creating a new menu. */
 export interface MenuOptions {
     /** Specify an id to use for the new menu. */
diff --git a/menu/menuItem.d.ts b/menu/menuItem.d.ts
index b57093a80119fc2e663351f9c1ced21cc47c2dbc..8c67989a29d99a2e138d65d7d5cf9552d8e2c606 100644
--- a/menu/menuItem.d.ts
+++ b/menu/menuItem.d.ts
@@ -1,4 +1,4 @@
-import { MenuItemBase } from './base';
+import { MenuItemBase } from './base.js';
 /** Options for creating a new menu item. */
 export interface MenuItemOptions {
     /** Specify an id to use for the new menu item. */
diff --git a/menu/predefinedMenuItem.d.ts b/menu/predefinedMenuItem.d.ts
index d30d66d3096172d130d2168e5e649948ffb4f596..6941ed17d13677c06feb40a21b5da586c0c6a5ad 100644
--- a/menu/predefinedMenuItem.d.ts
+++ b/menu/predefinedMenuItem.d.ts
@@ -1,5 +1,5 @@
-import { MenuItemBase } from './base';
-import { Image } from '../image';
+import { MenuItemBase } from './base.js';
+import { Image } from '../image.js';
 /** A metadata for the about predefined menu item. */
 export interface AboutMetadata {
     /** Sets the application name. */
diff --git a/menu/submenu.d.ts b/menu/submenu.d.ts
index 6e5ee8b3b9157657e864edad02cd319cdfe382ef..efb0b8d521022b181abd5046bc1842d488c92d99 100644
--- a/menu/submenu.d.ts
+++ b/menu/submenu.d.ts
@@ -1,12 +1,12 @@
-import { IconMenuItemOptions, PredefinedMenuItemOptions, CheckMenuItemOptions } from '../menu';
-import { MenuItem, type MenuItemOptions } from './menuItem';
-import { CheckMenuItem } from './checkMenuItem';
-import { IconMenuItem } from './iconMenuItem';
-import { PredefinedMenuItem } from './predefinedMenuItem';
-import { type LogicalPosition, PhysicalPosition, type Window } from '../window';
-import { type ItemKind, MenuItemBase } from './base';
-import { type MenuOptions } from './menu';
-import { MenuIcon } from '../image';
+import { IconMenuItemOptions, PredefinedMenuItemOptions, CheckMenuItemOptions } from '../menu.js';
+import { MenuItem, type MenuItemOptions } from './menuItem.js';
+import { CheckMenuItem } from './checkMenuItem.js';
+import { IconMenuItem } from './iconMenuItem.js';
+import { PredefinedMenuItem } from './predefinedMenuItem.js';
+import { type LogicalPosition, PhysicalPosition, type Window } from '../window.js';
+import { type ItemKind, MenuItemBase } from './base.js';
+import { type MenuOptions } from './menu.js';
+import { MenuIcon } from '../image.js';
 /** @ignore */
 export declare function itemFromKind([rid, id, kind]: [number, string, ItemKind]): Submenu | MenuItem | PredefinedMenuItem | CheckMenuItem | IconMenuItem;
 export type SubmenuOptions = (Omit<MenuItemOptions, 'accelerator' | 'action'> & MenuOptions) & {
diff --git a/menu.d.ts b/menu.d.ts
index 0aacba4d3a529ad6655381b96cdd6579b359d550..7379b0f40d300acb597b6ae4f9659ba8c1f16d8c 100644
--- a/menu.d.ts
+++ b/menu.d.ts
@@ -1,9 +1,9 @@
-export * from './menu/submenu';
-export * from './menu/menuItem';
-export * from './menu/menu';
-export * from './menu/checkMenuItem';
-export * from './menu/iconMenuItem';
-export * from './menu/predefinedMenuItem';
+export * from './menu/submenu.js';
+export * from './menu/menuItem.js';
+export * from './menu/menu.js';
+export * from './menu/checkMenuItem.js';
+export * from './menu/iconMenuItem.js';
+export * from './menu/predefinedMenuItem.js';
 /**
  * Menu types and utilities.
  *
diff --git a/mocks.d.ts b/mocks.d.ts
index 8eeb818dbc5b88bdf256ed6c7ad3be4c810ad7f0..6b5ad94ae5c8e186286e719de867bb57e59647b4 100644
--- a/mocks.d.ts
+++ b/mocks.d.ts
@@ -1,4 +1,4 @@
-import type { InvokeArgs } from './core';
+import type { InvokeArgs } from './core.js';
 /**
  * Options for `mockIPC`.
  *
diff --git a/tray.d.ts b/tray.d.ts
index 23780f46c8e2a1d34a3ab52ed8039c898f7b764a..43f7b07bc502b9d2a9181b430ebe8e4e5e0c8b95 100644
--- a/tray.d.ts
+++ b/tray.d.ts
@@ -1,7 +1,7 @@
-import type { Menu, Submenu } from './menu';
-import { Resource } from './core';
-import { Image } from './image';
-import { PhysicalPosition, PhysicalSize } from './dpi';
+import type { Menu, Submenu } from './menu.js';
+import { Resource } from './core.js';
+import { Image } from './image.js';
+import { PhysicalPosition, PhysicalSize } from './dpi.js';
 export type MouseButtonState = 'Up' | 'Down';
 export type MouseButton = 'Left' | 'Right' | 'Middle';
 export type TrayIconEventType = 'Click' | 'DoubleClick' | 'Enter' | 'Move' | 'Leave';
diff --git a/webview.d.ts b/webview.d.ts
index 787cc8662249a055d4e1372f4353c88ffc770236..eddb5defd5367d06635e618004799761201a00ee 100644
--- a/webview.d.ts
+++ b/webview.d.ts
@@ -11,13 +11,13 @@
  *
  * @module
  */
-import { PhysicalPosition, PhysicalSize } from './dpi';
-import type { LogicalPosition, LogicalSize } from './dpi';
-import { Position, Size } from './dpi';
-import type { EventName, EventCallback, UnlistenFn } from './event';
-import { type EventTarget } from './event';
-import { BackgroundThrottlingPolicy, ScrollBarStyle, Color, Window } from './window';
-import { WebviewWindow } from './webviewWindow';
+import { PhysicalPosition, PhysicalSize } from './dpi.js';
+import type { LogicalPosition, LogicalSize } from './dpi.js';
+import { Position, Size } from './dpi.js';
+import type { EventName, EventCallback, UnlistenFn } from './event.js';
+import { type EventTarget } from './event.js';
+import { BackgroundThrottlingPolicy, ScrollBarStyle, Color, Window } from './window.js';
+import { WebviewWindow } from './webviewWindow.js';
 /** The drag and drop event types. */
 type DragDropEvent = {
     type: 'enter';
diff --git a/webviewWindow.d.ts b/webviewWindow.d.ts
index f5ae32c6d406d26bd340dcd4c81b81c25266af67..e9a04d0793c9c190ee1e21b393d9481f487e1f7d 100644
--- a/webviewWindow.d.ts
+++ b/webviewWindow.d.ts
@@ -1,8 +1,8 @@
-import { Webview, WebviewLabel, WebviewOptions } from './webview';
-import type { WindowOptions } from './window';
-import { Window } from './window';
-import type { EventName, EventCallback, UnlistenFn } from './event';
-import type { Color, DragDropEvent } from './webview';
+import { Webview, WebviewLabel, WebviewOptions } from './webview.js';
+import type { WindowOptions } from './window.js';
+import { Window } from './window.js';
+import type { EventName, EventCallback, UnlistenFn } from './event.js';
+import type { Color, DragDropEvent } from './webview.js';
 /**
  * Get an instance of `Webview` for the current webview window.
  *
diff --git a/window.d.ts b/window.d.ts
index 641e46715cf607caefaff19327bfdb7c89f17c50..d280bb835483c5fd272a9456fd9bec6f4aaf324e 100644
--- a/window.d.ts
+++ b/window.d.ts
@@ -11,12 +11,12 @@
  *
  * @module
  */
-import { LogicalPosition, LogicalSize, PhysicalPosition, PhysicalSize, Position, Size } from './dpi';
-import type { Event, EventName, EventCallback, UnlistenFn } from './event';
-import { type EventTarget } from './event';
-import { WebviewWindow } from './webviewWindow';
-import type { DragDropEvent } from './webview';
-import { Image } from './image';
+import { LogicalPosition, LogicalSize, PhysicalPosition, PhysicalSize, Position, Size } from './dpi.js';
+import type { Event, EventName, EventCallback, UnlistenFn } from './event.js';
+import { type EventTarget } from './event.js';
+import { WebviewWindow } from './webviewWindow.js';
+import type { DragDropEvent } from './webview.js';
+import { Image } from './image.js';
 /**
  * Allows you to retrieve information about a given monitor.
  *
