diff --git a/dist/index.d.ts b/dist/index.d.ts
index 60be742e87bc4a1b5cdb2f8ff74360bfc7845443..03e265c60b0d3225c241aa10b58b663361432ba1 100644
--- a/dist/index.d.ts
+++ b/dist/index.d.ts
@@ -1,4 +1,4 @@
 export {}
-import VitestFailOnConsole from './src/index'
+import VitestFailOnConsole from './src/index.js'
 export default VitestFailOnConsole
 export {}
diff --git a/dist/src/index.d.ts b/dist/src/index.d.ts
index 11e201ca150cd6906deeee7252ff0c8e410bc8f9..1326ae14790268ac44fe24643a5063c1e7b77b02 100644
--- a/dist/src/index.d.ts
+++ b/dist/src/index.d.ts
@@ -1,3 +1,3 @@
-import { VitestFailOnConsoleFunction } from './types';
+import { VitestFailOnConsoleFunction } from './types.js';
 declare const init: ({ errorMessage, shouldFailOnAssert, shouldFailOnDebug, shouldFailOnError, shouldFailOnInfo, shouldFailOnLog, shouldFailOnWarn, skipTest, allowMessage, silenceMessage, afterEachDelay, shouldPrintMessage }?: VitestFailOnConsoleFunction) => void;
 export default init;
